### PR TITLE
PCA - Close Notifications

### PR DESCRIPTION
**Pull request by andyjiang3**

### CODE REVIEW

File: frontend/alert/components/common/DropdownTool.tsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    actionsText: String[];
+    functions: (() => void)[];
+    width: string;
+    img: string;
+}
+
+const DropdownTool = ({
+    actionsText,
+    functions,
+    width,
```
Comment (andyjiang3) [inline frontend/alert/components/common/DropdownTool.tsx:75]: Why do we need to pass in `width`? Can we change `DropdownMenu`'s width to auto and add some horizontal padding to `DropdownItem`

```
const DropdownItem = styled.div`
    padding: 0.25rem 0.5rem;
    &:hover {
        background-color: #e1e3e7;
    }
`;
```

then, we can change the styling of the parent div of `DropdownMenu` to center the dropdown:
```
<div
      style={{
          position: "relative",
          display: "flex",
          justifyContent: "center",
      }}
>
```
Maybe we can also turn this div into a styled component in this case.

File: frontend/alert/components/managealert/ToggleManager.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import { AlertAction, WrappedStyled } from "../../types";
+import ToggleSwitch from "../common/ToggleSwitch";
+
+type ActionFlexProps = FlexProps & {
+    background: string;
+};
+
+// Component associated with Resubscribe and Cancel buttons
+// for each alert
+interface ToggleManagerProps {
```
Comment (andyjiang3) [inline frontend/alert/components/managealert/ToggleManager.tsx:14]: nit: I think this comment is outdated since we removed resubscribe and cancel buttons.

**Review by andyjiang3 — COMMENTED** [review]
Really awesome work with this PR! Left some nits and code quality suggestions. Also, there seems to be a larger gap between the status and subscription column, is it possible to reduce this? Also curious if it would be better to replace the trash icon with a x icon, it kinda seems too big and out of place.

Screen Shot 2022-09-12 at 3 19 49 PM

File: frontend/alert/components/managealert/Header.tsx [hunk 3]
```diff
@@ -75,67 +122,75 @@ const Header = ({
...
+                                            AlertAction.ONCLOSED
+                                        ),
+                                    () =>
+                                        batchActionHandler(
+                                            AlertAction.OFFCLOSED
+                                        ),
+                                ]}
+                                width={"10.5"}
+                                img={"/svg/bell.svg"}
+                            />
```
[the remark this answers is not in the document]
Comment (mialana) [reply frontend/alert/components/managealert/Header.tsx:175]: Does using img cause that much of a problem? If we want to use DropdownTool for other purposes in the future, why not just leave img for now?

File: frontend/alert/components/AlertForm/index.tsx [hunk 4]
```diff
@@ -3,6 +3,8 @@ import PropTypes from "prop-types";
 import styled from "styled-components";
 import * as Sentry from "@sentry/browser";
 
+import InfoTool from "pcx-shared-components/src/common/InfoTool";
+
```
[the remark this answers is not in the document]
Comment (andyjiang3) [reply frontend/alert/components/AlertForm/index.tsx:7]: wait nvm for this

File: frontend/alert/components/managealert/Header.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (andyjiang3) [reply frontend/alert/components/managealert/Header.tsx:175]: Yeah I think definitely either way is fine.

File: frontend/alert/components/AlertForm/index.tsx [hunk 5]
```diff
@@ -191,54 +209,46 @@ const AlertForm = ({
...
-            })
+            postRegistration(autoCompleteInputRef.current.value)
+                .then((res) => {
+                    if (res.ok) {
+                        clearInputValue();
+                        setClosedNotif(false);
+                    }
+                    setResponse(res);
+                })
+                .catch(handleError);
```
[the remark this answers is not in the document]
Comment (mialana) [reply frontend/alert/components/AlertForm/index.tsx:253]: Andy, not sure what you mean by the two comments about isCourseOpen. Can you clarify briefly?

**Comment by mialana** [comment]
Thanks for the comments Andy! I looked over all of them and made changes as needed. Let me know if you see anything else I should fix. Thanks!

**Review by mialana — COMMENTED** [review]
Thanks for the comments Andy! I looked over all of them and made changes as needed. Let me know if you see anything else I should fix. Thanks!

**Comment by andyjiang3** [comment]
Besides from the weird behavior I mentioned on slack, I think everything looks good! Gonna merge for now and push a fix later if the behavior turns out to be a bug.

**Pull request merged**